vhd/lib/libvhd-index.c

```diff
@@ -682,13 +682,14 @@ vhdi_append_block(vhdi_context_t *ctx, vhdi_block_t *block, uint32_t *sector)
 }
 
 static int
-vhdi_copy_path_to(vhdi_path_t *path, const char *src, const char *dest)
+vhdi_copy_path_to(vhdi_path_t *path, const char *src, const char *dest, size_t dest_size)
 {
 	int len, err;
 	char *file, *relative_path, copy[VHD_MAX_NAME_LEN];
 	char *absolute_path, __absolute_path[PATH_MAX];
 
-	strcpy(copy, dest);
+	strncpy(copy, dest, dest_size);
```

Comment:
`dest` is a source here not a destination, so the size here should be `sizeof(copy)`, otherwise a very large destsize could still cause a stack overwrite.